tests/test_fts3_transfer_tools.py

```diff
@@ -0,0 +1,397 @@
+
+import pytest
+import logging
+
+from rucio.transfertool.fts3 import FTS3Transfertool, build_job_params
+from rucio.core.topology import Topology
+from rucio.core.transfer import ProtocolFactory, build_transfer_paths
+from rucio.core.request import list_and_mark_transfer_requests_and_source_replicas
+from rucio.common.config import (
+    config_set,
+    config_add_section,
+    config_get,
+    config_remove_option,
+)
+from rucio.db.sqla.session import get_session
+from rucio.daemons.conveyor.submitter import submitter
+from rucio.core.plugins import PolicyPackageAlgorithms
+from rucio.transfertool.fts3_plugins import FTS3SchedulingHints, FTS3CollocationHints
+
+from rucio.core import distance as distance_core
+from rucio.core import replica as replica_core
+from rucio.core import rule as rule_core
+
+
+mock_session = get_session()
+
+MAX_POLL_WAIT_SECONDS = 60
+TEST_FTS_HOST = "https://fts:8446"
+
+config_add_section("transfers")
+config_add_section("tape_prority")
+
+
+class FTS3CollocationHintsMock(FTS3CollocationHints):
+    def __init__(self):
+        super().__init__()
+
+    def set_collocation(self, *args, **kwargs):
+        return {"0": "", "1": "", "2": "", "3": ""}
+
+
+def _make_config_section():
+    try:
+        config_add_section("transfers")
+        config_add_section("tape_prority")
+    except:
+        pass
+
+
+def _make_transfer_path(did, rse_factory, root_account):
+    _, src_rse_id = rse_factory.make_mock_rse()
+    dst_rse, dst_rse_id = rse_factory.make_mock_rse()
+    all_rses = [src_rse_id, dst_rse_id]
+    distance_core.add_distance(src_rse_id, dst_rse_id, distance=1)
+
+    topology = Topology(rse_ids=all_rses)
+    file = {"scope": did["scope"], "name": did["name"], "bytes": 1}
+    replica_core.add_replicas(rse_id=src_rse_id, files=[file], account=root_account)
+    rule_core.add_rule(
+        dids=[did],
+        account=root_account,
+        copies=1,
+        rse_expression=dst_rse,
+        grouping="ALL",
+        weight=None,
+        lifetime=None,
+        locked=False,
+        subscription_id=None,
+    )
+
+    requests_by_id = list_and_mark_transfer_requests_and_source_replicas(
+        rse_collection=topology, rses=[src_rse_id, dst_rse_id]
+    )
+    requests, *_ = build_transfer_paths(
+        topology=topology,
+        protocol_factory=ProtocolFactory(),
+        requests_with_sources=requests_by_id.values(),
+    )
+    _, [transfer_path] = next(iter(requests.items()))
+
+    return transfer_path
+
+
+def test_scheduling_hints(did_factory, rse_factory, root_account):
+
+    """Transfer with an installed plugin for scheduling based on transfer activity"""
+
+    # Produce a new did
+    mock_did = did_factory.random_file_did()
+
+    # Make the transfer path
+    transfer_path = _make_transfer_path(mock_did, rse_factory, root_account)
+
+    activity = transfer_path[0].rws.activity
+    _make_config_section()
+    config_set(section="tape_prority", option=activity, value="100")
```

Comment:
Setting config values in tests like that can impact all other tests running in parallel. Check file_config_mock and core_config_mock fixtures.

Reply:
Ah, didn't know about those. Changing it now.